flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/tieredstore/upstream/common/StorageTier.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.partition.tieredstore.upstream.common;
+
+import org.apache.flink.runtime.io.network.partition.BufferAvailabilityListener;
+
+import java.io.IOException;
+
+/**
+ * The single storage tier for the tiered store. The storage tiers is independent of each other.
+ * Through the {@link StorageTier}, we can create {@link TierWriter} to write shuffle data and
+ * create the {@link TierReaderView} to consume shuffle data.
+ */
+public interface StorageTier {
+
+    void setup() throws IOException;
+
+    /** Create the {@link TierWriter} of the {@link StorageTier} to write shuffle data. */
+    TierWriter createPartitionTierWriter();
+
+    /** Create the {@link TierReaderView} of the {@link StorageTier} to read shuffle data. */
+    TierReaderView createTierReaderView(
+            int subpartitionId, BufferAvailabilityListener availabilityListener) throws IOException;
+
+    /**
+     * Before start a new segment, the writer side calls the method to check whether the {@link
+     * StorageTier} can store the next segment.
+     */
+    boolean canStoreNextSegment(int subpartitionId);
```

Comment:
Use tiered store language. `subpartitionId` can be replaced by `producerId`.

Reply:
The method has been removed from the API. It will be in the netty-based consumer API in the subsequential change. 

In that change, we will use the `producerId`  instead.